app/src/marketplace/containers/ProductPage/PurchaseModal.jsx

```diff
@@ -287,77 +298,69 @@ export const PurchaseDialog = ({ productId, api }: Props) => {
     const onSetAccessPeriod = useCallback(async (accessPeriod: AccessPeriod) => {
         accessPeriodParams.current = accessPeriod
 
-        purchasePrice.current = dataForTimeUnits(
-            pricePerSecond,
-            dataPerUsd,
-            priceCurrency,
-            accessPeriod.time,
-            accessPeriod.timeUnit,
-        )
-
         if (accountLinked) {
             onVerifyAllowance()
         } else {
             setStep(purchaseFlowSteps.LINK_ACCOUNT)
         }
-    }, [accountLinked, pricePerSecond, dataPerUsd, onVerifyAllowance, priceCurrency])
+    }, [accountLinked, onVerifyAllowance])
 
     const onSetDataAllowance = useCallback(async () => {
-        if (!accessPeriodParams.current || !purchasePrice.current) {
+        if (!accessPeriodParams.current) {
             throw new Error(I18n.t('error.noProductOrAccess'))
         }
 
-        const { paymentCurrency } = accessPeriodParams.current
+        const { paymentCurrency, price } = accessPeriodParams.current
 
         try {
-            await validateBalanceForPurchase(purchasePrice.current, paymentCurrency)
+            await validateBalanceForPurchase({
+                price,
+                paymentCurrency,
+            })
 
             if (!isMounted()) { return }
 
             if (BN(dataAllowance).isGreaterThan(0)) {
                 await dispatch(resetDataAllowanceToContract())
             } else {
-                await dispatch(setDataAllowanceToContract((purchasePrice.current || 0).toString()))
+                await dispatch(setDataAllowanceToContract(price))
             }
         } catch (e) {
             setPurchaseError(e)
         }
     }, [isMounted, dataAllowance, dispatch])
 
     const onSetDaiAllowance = useCallback(async () => {
-        if (!accessPeriodParams.current || !purchasePrice.current) {
+        if (!accessPeriodParams.current) {
             throw new Error(I18n.t('error.noProductOrAccess'))
         }
 
-        const { paymentCurrency, priceInDai } = accessPeriodParams.current
+        const { paymentCurrency, price } = accessPeriodParams.current
 
         try {
-            await validateBalanceForPurchase(purchasePrice.current, paymentCurrency)
+            await validateBalanceForPurchase({
```

Comment:
this `validateBalanceForPurchase` should be enough to call once but left for now